### PR TITLE
1D solver API docs

### PR DESCRIPTION
**Pull request by speth**
**Changes proposed in this pull request**

- Document all member functions and member variables of the C++ classes comprising the 1D solver
- Deprecate/remove a handful of redundant/unused methods and variables in 1D classes

**If applicable, fill in the issue number this pull request is fixing**

This is the last step needed to complete Cantera/enhancements#25.

**Checklist**

- [x] The pull request includes a clear description of this code change
- [x] Commit messages have short titles and reference relevant issues
- [x] Build passes (`scons build` & `scons test`) and unit tests address code coverage
- [x] Style & formatting of contributed code follows [contributing guidelines](https://github.com/Cantera/cantera/blob/main/CONTRIBUTING.md)
- [x] The pull request is ready for review

### CODE REVIEW

File: include/cantera/oneD/Flow1D.h [hunk 1]
```diff
@@ -141,7 +149,7 @@ class Flow1D : public Domain1D
 
     void _finalize(const double* x) override;
 
-    //! Sometimes it is desired to carry out the simulation using a specified
+    //! Sometimes it is desired to carry out the simulation fusing a specified
```
Comment (ischoegl) [inline include/cantera/oneD/Flow1D.h]: ?

File: include/cantera/oneD/IonFlow.h [hunk 2]
```diff
@@ -28,9 +28,14 @@ namespace Cantera
 class IonFlow : public Flow1D
 {
 public:
+    //! Create a new IonFlow domain.
+    //! @param ph Object representing the gas phase. This object will be used
```
Comment (ischoegl) [inline include/cantera/oneD/IonFlow.h]: Nitpick: two spaces after parameter names makes things more readable … not a major issue overall.

File: include/cantera/oneD/MultiJac.h [hunk 3]
```diff
@@ -23,6 +23,8 @@ namespace Cantera
 class MultiJac : public BandMatrix
 {
 public:
+    //! Constructor.
+    //! @param r  The nonlinear system for which to compute the Jacobian.
```
Comment (ischoegl) [inline include/cantera/oneD/MultiJac.h:27]: This version uses two spaces

File: src/oneD/OneDim.cpp [hunk 4]
```diff
@@ -220,13 +216,13 @@ int OneDim::solve(double* x, double* xnew, int loglevel)
     return m_newt->solve(x, xnew, *this, *m_jac, loglevel);
 }
 
-void OneDim::evalSSJacobian(double* x, double* xnew)
+void OneDim::evalSSJacobian(double* x, double* r)
```
Comment (ischoegl) [inline src/oneD/OneDim.cpp]: I’m generally not convinced that single-letter variables or functions are a good idea.

File: include/cantera/oneD/Flow1D.h [hunk 5]
```diff
@@ -849,48 +891,99 @@ class Flow1D : public Domain1D
...
     vector<bool> m_do_energy;
+
+    //! `true` if the Soret diffusion term should be calculated.
     bool m_do_soret = false;
+
+    //! Determines whether diffusive fluxes are computed using gradients of mass
+    //! fraction or mole fraction.
+    //! @see setFluxGradientBasis, fluxGradientBasis
     ThermoBasis m_fluxGradientBasis = ThermoBasis::molar;
-    vector<bool> m_do_species;
```
Comment (ischoegl) [inline include/cantera/oneD/Flow1D.h:868]: It would be good if you could comment on the rationale for removing this feature.

**Review by ischoegl — COMMENTED** [review]
Thanks, @speth. I have a couple of minor comments.

File: include/cantera/oneD/Flow1D.h (hunk 1 above)
Comment (speth) [reply include/cantera/oneD/Flow1D.h]: Oops. Good catch.

File: src/oneD/OneDim.cpp (hunk 4 above)
Comment (speth) [reply src/oneD/OneDim.cpp]: I renamed it `rsd`, which is what's used a few other places, but just `r` is used in other places in the 1D solver for the residual. Here, `r` is strictly an improvement, as the old name was misleading -- this definitely does not contain a "new" value for `x`.

File: include/cantera/oneD/Flow1D.h (hunk 5 above)
Comment (speth) [reply include/cantera/oneD/Flow1D.h:868]: This was just a variable that had no methods for being set except when being restored from a saved data file, and was not being used to control the logic for evaluating the governing equations.

Digging through the Git history a bit jogged my memory. The idea of solving with a fixed species profile was a feature that was removed in Cantera 2.3, but I'm not sure it ever made sense from a consistency standpoint.

**Review by ischoegl — APPROVED** [review]
Thanks, @speth!

**Pull request merged**